### PR TITLE
Upgrade to OpenTelemetry 1.0

### PR DESCRIPTION
**Pull request by owais**
# Description

Upgraded to OpenTelemetry 1.0 packages. Also prepared release 0.10.0

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

- [x] Tested manually
- [x] Updated automated tests

# Checklist:

- [x] Changelogs have been updated
- [x] Unit tests have been added/updated
- [x] Documentation has been updated

### CODE REVIEW

File: splunk_otel/options.py [hunk 1]
```diff
@@ -56,10 +56,15 @@ def __init__(
...
+        self.access_token = access_token
 
         if not max_attr_length:
-            max_attr_length = splunk_env_var("MAX_ATTR_LENGTH", DEFAULT_MAX_ATTR_LENGTH)
+            try:
+                max_attr_length = int(
+                    splunk_env_var("MAX_ATTR_LENGTH", DEFAULT_MAX_ATTR_LENGTH)
+                )
+            except TypeError:
+                pass
```
Comment (pellared) [inline splunk_otel/options.py:67]: maybe it would be worth log the conversion error?

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,11 +2,17 @@
...
 
-- `init_tracing()` now accepts an argument called `endpoint` instead of `url`.
+- Upgraded OpenTelemetry to 1.0
 - `SPLUNK_TRACE_EXPORTER_URL` was replaced with `OTEL_EXPORTER_JAEGER_ENDPOINT`.
-- `start_tracing` now accepts `access_token` and `max_attr_length` options.
+- `SPLUNK_MAX_ATTR_LENGTH` properly supported now. This env var was documented
+  but due to a regression was not working anymore. This version adds proper
+  support for it.
+- `start_tracing()` now accepts `access_token` and `max_attr_length` options.
+- `start_tracing()` now accepts an argument called `endpoint` instead of `url`.
```
Comment (pellared) [inline CHANGELOG.md:15]: nit: maybe merge into one line?

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,11 +2,17 @@
 
 ## Unreleased
 
+## 0.10.0 (03-27-2021)
```
Comment (pellared) [inline CHANGELOG.md:5]: nit: I like when the Changlog have hyperlinks to releases/changes, PRs
it is even sometimes helpful during development/bug fixing (at least for me 😉 )

**Review by pellared — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (owais) [reply CHANGELOG.md:5]: Me too but it's annoying to do after the fact. If we let this pass, I'll make sure we make every PR include link to itself in changelog in future :)

Comment (owais) [reply CHANGELOG.md:5]: Actually updated it. It was just a handful of changes.

**Pull request merged**